crates/gpui/src/platform/linux/wayland/client.rs

```diff
@@ -567,9 +606,18 @@ impl Dispatch<wl_pointer::WlPointer, ()> for WaylandClientState {
         conn: &Connection,
         qh: &QueueHandle<Self>,
     ) {
+        let mut cursor_state = state.1.borrow_mut();
         let mut state = state.0.borrow_mut();
+
+        if cursor_state.cursor.is_none() {
+            cursor_state.cursor = Some(Cursor::new(&conn, &state.compositor, &qh, &state.shm, 24));
```

Comment:
I have set the cursor size to 24, which seems to generally be the default size. But there doesn't seem to be a universal way to know what is the cursor size used by DE. Gnome does it in its own way and kde in its own way. Maybe later we can make this configurable from the settings?

Reply:
At the very least I would love to see the cursor scaled correctly according to the scaling factor of whatever display it is on. I have one monitor set to 200% scaling and some apps are very awkward to use because the cursor becomes *tiny* when moving over them.

Reply:
Seems to work. I set 200% scaling in my monitor and switched back. The cursor size scaled accordingly. Here is a video of my screen recording:

[zed_cursor_wayland_scaling.webm](https://github.com/zed-industries/zed/assets/12579216/6a747356-2060-4cab-a090-98060204467a)

I think the existing wayland fractional scaling support helps here.

Reply:
Considering the cursor size scales automatically based on display scaling, making the cursor size configurable is not an immediate concern. I think we can work on that later. Marking as resolved for now. Feel free to discuss on this further if anyone has different opinion.